### PR TITLE
Move CI from Travis to GitHub Actions

### PR DESCRIPTION
**Pull request by dividedmind**
Note for deploy to work `NPM_TOKEN` needs to be provided as a secret.

### CODE REVIEW

File: components/error/default/index.mjs
Comment (lachrist) [inline components/error/default/index.mjs:49]: Same remark as for your previous dynamic error message. I'm not against improving the current logging mechanism (detailed dynamic log + short static error message) but I don't see the point in changing it in two places where there are like 100+ locations where it is used. In my opinion this should be discussed and changed in a separate PR so that we can group all these change together and retain some consistency.

**Review by lachrist — CHANGES_REQUESTED** [review]
Here are my three remarks:
- consider that `npm run coverage` does perform integration tests.
- I would move log changes to another PR.
- `fswin` seems heavy and polute the pure path component.

File: .github/workflows/ci.yml
Comment (dividedmind) [reply .github/workflows/ci.yml]: Good point, I missed that. I've split `test` into separate scripts in 614b0f73f36.

File: components/command/node/jest-config.mjs
Comment (dividedmind) [reply components/command/node/jest-config.mjs:46]: The problem is, turning logging on is another step which can cause friction and it provides a lot of potentially unnecessary information. I don't want to have to rerun CI and delve through all the log messages to figure out what the actual single error was. Current laconic error messages give the user zero indication of what actually went wrong. We don't need (and we shouldn't) give the user a stacktrace by default (although we actually do!), but I believe we should (and need to) at least tell the user more specifically want went wrong instead of a enigmatic generic message that leaves them puzzled and confused.

File: package.json
Comment (dividedmind) [reply package.json:49]: > Couple of questions/remarks:
> 
>     * Could you elaborate on when/how the short path shown its ugly head?

There are a couple of places where the code does sanity checks on code paths (for example [here](https://github.com/getappmap/appmap-agent-js/blob/e876f95881b1d8a7e66b32fca169794a585e1c2d/components/setup/node/index.mjs#L129)) and because it arrives at the path through different methods it can cause one to be short and the other to be long (see for example [this error](https://github.com/getappmap/appmap-agent-js/actions/runs/5500501136/jobs/10023447872#step:6:426)).

>     * Do we have to use this third-party package? I'm nervous at the idea of introducing a c++ depencency to solve what seems to be a minor problem.
>     * The path component should remain pure. We should either provide the information necessary to `toDirectoryPath` solve short paths in a pure manner. Or solve them at the callcite.

We have to call the system API; this is the only way to resolve a short path from a long one and vice-versa. It's not possible to do it in a pure way since the path mapping is stored in the filesystem and the only way to access it is through this API. And fswin is just a very thin wrapper on the relevant APIs. The alternative would be to write it ourselves :)

File: components/error/default/index.mjs
Comment (dividedmind) [reply components/error/default/index.mjs:49]: Like I said before, I think short error message needs to contain some specific information to be actually useful. Otherwise it's just annoying and useless. But I agree that it's better to strategically improve all the sites that can benefit from it. I just put these in as I needed to solve the problems I encountered.

OTOH, I don't want this to become another rabbit hole we spend too much time on while we have other issues to deal with. Now I think maybe it's okay to keep those tactical changes and add an issue to address this problem more systematically. If we find the time and impetus to do it while doing some other work, even better, but perhaps we shouldn't take out those tactical changes just for the sake of purity; if I encountered these, perhaps the users will too.

**Comment by dividedmind** [comment]
Can you take another look? Obviously please don't merge as-is, I need to rebase the fixups, I'll take care of it before merging.

> Here are my three remarks:
> 
>     * consider that `npm run coverage` does perform integration tests.

Fixed.

>     * I would move log changes to another PR.

Thinking about it some more, I think we should keep the tactical changes and open an issue for a more strategic treatment on this. We need to keep moving forward :)

>     * `fswin` seems heavy and polute the pure path component.

It's the opposite of heavy, it's a dependency-less thin wrapper on the relevant OS APIs. Perhaps it does pollute the path component a bit, but I don't see much of a way around it without needlessly complicating the code. Every abstraction is leaky in the end :)

**Comment by brikelly** [comment]
@lachrist can you please respond to @dividedmind's comments?

**Pull request merged**
